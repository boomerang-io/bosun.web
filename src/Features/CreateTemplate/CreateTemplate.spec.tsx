import React from "react";
import Inputs from "../CreateTemplate";

const mockfn = jest.fn();

const props = {
  loading: false,
  updateInputs: mockfn,
  workflowActions: { deleteWorkflowInput: mockfn },
};

beforeEach(() => {
  document.body.setAttribute("id", "app");
});

describe("Inputs --- Snapshot Test", () => {
  it("Capturing Snapshot of Inputs", () => {
    const { baseElement } = rtlRender(<Inputs {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
