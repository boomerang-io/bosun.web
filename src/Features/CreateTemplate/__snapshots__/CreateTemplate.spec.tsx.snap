// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inputs --- Snapshot Test Capturing Snapshot of Inputs 1`] = `
<body
  id="app"
>
  <div>
    <form
      action="#"
    >
      <header
        class="bx--bmrg-feature-header"
      >
        <section
          class="bx--bmrg-feature-header__content"
        >
          <div
            class="bx--bmrg-feature-header__header"
          >
            <h1
              class="bx--bmrg-feature-header-title"
            >
              Create Template
            </h1>
            
          </div>
        </section>
        <section
          class="buttons"
        >
          <button
            class="button bx--btn bx--btn--field bx--btn--secondary"
            tabindex="0"
            type="button"
          >
            Cancel
          </button>
          <button
            class="button bx--btn bx--btn--field bx--btn--primary"
            data-testid="policy-header-affirmative-action"
            tabindex="0"
            type="submit"
          >
            Create
            <svg
              aria-hidden="true"
              aria-label="Create"
              class="bx--btn__icon"
              fill="currentColor"
              focusable="false"
              height="16"
              preserveAspectRatio="xMidYMid meet"
              role="img"
              viewBox="0 0 32 32"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
              />
            </svg>
          </button>
        </section>
      </header>
      <section
        class="container"
      >
        <div
          class="bx--tabs--scrollable"
        >
          <div
            class="bx--tabs--scrollable"
          >
            <button
              class="bx--tab--overflow-nav-button--hidden"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 8L10 3 10.7 3.7 6.4 8 10.7 12.3 10 13z"
                />
              </svg>
            </button>
            <ul
              class="bx--tabs--scrollable__nav"
              role="tablist"
            >
              <li
                aria-controls="undefined__panel"
                aria-selected="true"
                class="bx--tabs--scrollable__nav-item bx--tabs__nav-item--selected bx--tabs--scrollable__nav-item--selected"
                role="tab"
                tabindex="-1"
              >
                <a
                  class="bx--tabs--scrollable__nav-link"
                  href="#"
                  tabindex="0"
                >
                  About
                </a>
              </li>
              <li
                aria-controls="undefined__panel"
                aria-selected="false"
                class="bx--tabs--scrollable__nav-item"
                role="tab"
                tabindex="-1"
              >
                <a
                  class="bx--tabs--scrollable__nav-link"
                  href="#"
                  tabindex="-1"
                >
                  Rules
                </a>
              </li>
              <li
                aria-controls="undefined__panel"
                aria-selected="false"
                class="bx--tabs--scrollable__nav-item"
                role="tab"
                tabindex="-1"
              >
                <a
                  class="bx--tabs--scrollable__nav-link"
                  href="#"
                  tabindex="-1"
                >
                  OPA Rego
                </a>
              </li>
            </ul>
            <button
              class="bx--tab--overflow-nav-button--hidden"
              type="button"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                focusable="false"
                height="16"
                preserveAspectRatio="xMidYMid meet"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M11 8L6 13 5.3 12.3 9.6 8 5.3 3.7 6 3z"
                />
              </svg>
            </button>
          </div>
          <div
            aria-hidden="false"
            aria-live="polite"
            class="bx--tab-content"
            role="tabpanel"
          >
            <div
              class="generalContainer"
            >
              <section
                class="generalSection"
              >
                <h1
                  class="sectionTitle"
                >
                  General
                </h1>
                <div
                  class="bx--bmrg-text-input"
                >
                  <div
                    class="bx--form-item bx--text-input-wrapper"
                  >
                    <label
                      class="bx--label"
                      for="name"
                    >
                      <div
                        class="bx--bmrg-text-input__label"
                      >
                        <div>
                          Name
                        </div>
                      </div>
                    </label>
                    <div
                      class="bx--text-input__field-outer-wrapper"
                    >
                      <div
                        class="bx--text-input__field-wrapper"
                      >
                        <input
                          class="bx--text-input bx--text__input"
                          id="name"
                          name="name"
                          placeholder="Name"
                          title="Name"
                          type="text"
                          value=""
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="bx--bmrg-text-input"
                >
                  <div
                    class="bx--form-item bx--text-input-wrapper"
                  >
                    <label
                      class="bx--label"
                      for="key"
                    >
                      <div
                        class="bx--bmrg-text-input__label"
                      >
                        <div>
                          Key
                        </div>
                      </div>
                    </label>
                    <div
                      class="bx--text-input__field-outer-wrapper"
                    >
                      <div
                        class="bx--text-input__field-wrapper"
                      >
                        <input
                          class="bx--text-input bx--text__input"
                          id="key"
                          name="key"
                          placeholder="Key"
                          title="Key"
                          type="text"
                          value=""
                        />
                      </div>
                      <div
                        class="bx--form__helper-text"
                      >
                        Must match the OPA Rego package declaration
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="bx--bmrg-text-area"
                >
                  <div
                    class="bx--form-item"
                  >
                    <label
                      class="bx--label"
                      for="description"
                    >
                      <div
                        style="display: flex;"
                      >
                        <div>
                          Description (optional)
                        </div>
                      </div>
                    </label>
                    <div
                      class="bx--text-area__wrapper"
                    >
                      <textarea
                        aria-placeholder="Description"
                        class="bx--text-area"
                        cols="50"
                        id="description"
                        name="description"
                        placeholder="Description"
                        rows="4"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="bx--bmrg-text-input"
                >
                  <div
                    class="bx--form-item bx--text-input-wrapper"
                  >
                    <label
                      class="bx--label"
                      for="order"
                    >
                      <div
                        class="bx--bmrg-text-input__label"
                      >
                        <div>
                          Order (optional)
                        </div>
                      </div>
                    </label>
                    <div
                      class="bx--text-input__field-outer-wrapper"
                    >
                      <div
                        class="bx--text-input__field-wrapper"
                      >
                        <input
                          class="bx--text-input bx--text__input"
                          id="order"
                          name="order"
                          placeholder="0"
                          title="0"
                          type="number"
                          value="0"
                        />
                      </div>
                      <div
                        class="bx--form__helper-text"
                      >
                        Specify presentation order in Create Policy feature
                      </div>
                    </div>
                  </div>
                </div>
              </section>
              <section
                class="generalSection"
              >
                <h1
                  class="sectionTitle"
                >
                  Validation
                </h1>
                <div
                  class="bx--bmrg-creatable"
                >
                  <div
                    class="bx--bmrg-creatable__input"
                  >
                    <div
                      class="bx--form-item bx--text-input-wrapper"
                    >
                      <label
                        class="bx--label"
                        for="labels"
                      >
                        <div
                          style="display: flex;"
                        >
                          <div>
                            Labels (optional)
                          </div>
                        </div>
                      </label>
                      <div
                        class="bx--text-input__field-outer-wrapper"
                      >
                        <div
                          class="bx--text-input__field-wrapper"
                        >
                          <input
                            class="bx--text-input bx--text__input"
                            id="labels"
                            maxlength="64"
                            placeholder="Create labels"
                            title="Create labels"
                            type="text"
                            value=""
                          />
                        </div>
                        <div
                          class="bx--form__helper-text"
                        >
                          Metadata to pass information into the pre-integrated repositories
                        </div>
                      </div>
                    </div>
                    <button
                      class="bx--bmrg-creatable__button bx--btn bx--btn--field bx--btn--primary bx--btn--disabled"
                      disabled=""
                      tabindex="0"
                      type="button"
                    >
                      Add
                      <svg
                        aria-hidden="true"
                        aria-label="Add"
                        class="bx--btn__icon"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        role="img"
                        viewBox="0 0 32 32"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                        />
                      </svg>
                    </button>
                  </div>
                  <div
                    class="bx--bmrg-creatable__tags"
                  />
                </div>
              </section>
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-live="polite"
            class="bx--tab-content"
            hidden=""
            role="tabpanel"
          >
            <p>
              Create at least one
            </p>
            <div
              class="propertyList"
            >
              <button
                class="createPropertyCard"
                data-testid="create-new-workflow-input-button"
                type="button"
              >
                <div
                  class="createContainer"
                >
                  <svg
                    aria-label="Add"
                    class="createIcon"
                    fill="currentColor"
                    focusable="false"
                    height="32"
                    preserveAspectRatio="xMidYMid meet"
                    role="img"
                    viewBox="0 0 32 32"
                    width="32"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M17 15L17 8 15 8 15 15 8 15 8 17 15 17 15 24 17 24 17 17 24 17 24 15z"
                    />
                  </svg>
                  <p
                    class="createText"
                  >
                    Create a new rule
                  </p>
                </div>
              </button>
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-live="polite"
            class="bx--tab-content"
            hidden=""
            role="tabpanel"
          >
            <section
              class="opaPolicyContainer"
            >
              <div
                style="max-width: 80rem; max-height: 50rem; height: 100%; width: 100%; margin: auto; overflow-x: visible; display: flex; flex-direction: column; justify-content: flex-start;"
              >
                <div
                  class="bx--toolbar b-task-text-area"
                >
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-end"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Undo
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Undo"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 20 20"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15,9.3c1.5,0,2.8,1.3,2.8,2.7c0,1.5-1.3,2.7-2.8,2.7h-4V16h4c2.3,0,4-1.7,4-4c0-2.2-1.8-4-4-4H3.4	l2.5-2.5L5,4.6l-4,4l4,4l0.9-0.9L3.4,9.3H15z"
                      />
                    </svg>
                  </button>
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Redo
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Redo"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 20 20"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M5,9.3c-1.5,0-2.8,1.3-2.8,2.7c0,1.5,1.3,2.7,2.8,2.7h4V16H5c-2.3,0-4-1.7-4-4c0-2.2,1.8-4,4-4h11.6	l-2.5-2.5L15,4.6l4,4l-4,4l-0.9-0.9l2.5-2.5H5z"
                      />
                    </svg>
                  </button>
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Copy
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Copy"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M28,10V28H10V10H28m0-2H10a2,2,0,0,0-2,2V28a2,2,0,0,0,2,2H28a2,2,0,0,0,2-2V10a2,2,0,0,0-2-2Z"
                      />
                      <path
                        d="M4,18H2V4A2,2,0,0,1,4,2H18V4H4Z"
                      />
                    </svg>
                  </button>
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Cut
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Cut"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26.5,19.63,20.24,16l6.26-3.63a5,5,0,0,0-1.21-9.2A5.19,5.19,0,0,0,24,3a5,5,0,0,0-4.33,7.53,5,5,0,0,0,2.39,2.1l-3.82,2.21L4,6.6,3,8.34,16.24,16,3,23.68l1,1.74,14.24-8.26,3.82,2.21a5,5,0,0,0-2.39,2.1A5,5,0,0,0,24,29a5.19,5.19,0,0,0,1.29-.17,5,5,0,0,0,1.21-9.2ZM21.4,9.53a3,3,0,0,1,1.1-4.12,3,3,0,0,1,4.1,1.11,3,3,0,0,1-1.1,4.11h0A3,3,0,0,1,21.4,9.53Zm5.2,16a3,3,0,0,1-4.1,1.11,3,3,0,0,1-1.1-4.12,3,3,0,0,1,4.1-1.1h0A3,3,0,0,1,26.6,25.48Z"
                      />
                    </svg>
                  </button>
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Paste
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Paste"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="20"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 32 32"
                      width="20"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M26,20H17.83l2.58-2.59L19,16l-5,5,5,5,1.41-1.41L17.83,22H26v8h2V22A2,2,0,0,0,26,20Z"
                      />
                      <path
                        d="M23.71,9.29l-7-7A1,1,0,0,0,16,2H6A2,2,0,0,0,4,4V28a2,2,0,0,0,2,2h8V28H6V4h8v6a2,2,0,0,0,2,2h6v2h2V10A1,1,0,0,0,23.71,9.29ZM16,4.41,21.59,10H16Z"
                      />
                    </svg>
                  </button>
                  <div
                    aria-labelledby="search-search"
                    class="bx--search bx--search--sm"
                    role="search"
                  >
                    <svg
                      aria-hidden="true"
                      class="bx--search-magnifier"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M15,14.3L10.7,10c1.9-2.3,1.6-5.8-0.7-7.7S4.2,0.7,2.3,3S0.7,8.8,3,10.7c2,1.7,5,1.7,7,0l4.3,4.3L15,14.3z M2,6.5	C2,4,4,2,6.5,2S11,4,11,6.5S9,11,6.5,11S2,9,2,6.5z"
                      />
                    </svg>
                    <label
                      class="bx--label"
                      for="search"
                      id="search-search"
                    >
                      Search
                    </label>
                    <input
                      autocomplete="off"
                      class="bx--search-input"
                      id="search"
                      placeholder="Search"
                      role="searchbox"
                      type="text"
                    />
                    <button
                      aria-label=""
                      class="bx--search-close bx--search-close--hidden"
                      type="button"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        focusable="false"
                        height="16"
                        preserveAspectRatio="xMidYMid meet"
                        viewBox="0 0 32 32"
                        width="16"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M24 9.4L22.6 8 16 14.6 9.4 8 8 9.4 14.6 16 8 22.6 9.4 24 16 17.4 22.6 24 24 22.6 17.4 16 24 9.4z"
                        />
                      </svg>
                    </button>
                  </div>
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Find previous
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Find previous"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M3.7 6.7L7.5 2.9 7.5 15 8.5 15 8.5 2.9 12.3 6.7 13 6 8 1 3 6z"
                      />
                    </svg>
                  </button>
                  <button
                    class="b-task-text-area__button bx--btn bx--btn--sm bx--btn--ghost bx--btn--icon-only bx--tooltip__trigger bx--tooltip--a11y bx--tooltip--bottom bx--tooltip--align-center"
                    tabindex="0"
                    type="button"
                  >
                    <span
                      class="bx--assistive-text"
                    >
                      Find next
                    </span>
                    <svg
                      aria-hidden="true"
                      aria-label="Find next"
                      class="bx--btn__icon"
                      fill="currentColor"
                      focusable="false"
                      height="16"
                      preserveAspectRatio="xMidYMid meet"
                      role="img"
                      viewBox="0 0 16 16"
                      width="16"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M12.3 9.3L8.5 13.1 8.5 1 7.5 1 7.5 13.1 3.7 9.3 3 10 8 15 13 10z"
                      />
                    </svg>
                  </button>
                </div>
                <div
                  class="react-codemirror2"
                >
                  <div
                    class="CodeMirror cm-s-material CodeMirror-wrap"
                  >
                    <div
                      style="overflow: hidden; position: relative; width: 3px; height: 0px;"
                    >
                      <textarea
                        autocapitalize="off"
                        autocorrect="off"
                        spellcheck="false"
                        style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"
                        tabindex="0"
                      />
                    </div>
                    <div
                      class="CodeMirror-vscrollbar"
                      cm-not-content="true"
                      tabindex="-1"
                    >
                      <div
                        style="min-width: 1px;"
                      />
                    </div>
                    <div
                      class="CodeMirror-hscrollbar"
                      cm-not-content="true"
                      tabindex="-1"
                    >
                      <div
                        style="height: 100%; min-height: 1px;"
                      />
                    </div>
                    <div
                      class="CodeMirror-scrollbar-filler"
                      cm-not-content="true"
                    />
                    <div
                      class="CodeMirror-gutter-filler"
                      cm-not-content="true"
                    />
                    <div
                      class="CodeMirror-scroll"
                      tabindex="-1"
                    >
                      <div
                        class="CodeMirror-sizer"
                        style="margin-left: 0px;"
                      >
                        <div
                          style="position: relative;"
                        >
                          <div
                            class="CodeMirror-lines"
                            role="presentation"
                          >
                            <div
                              role="presentation"
                              style="position: relative; outline: none;"
                            >
                              <div
                                class="CodeMirror-measure"
                              >
                                <pre
                                  class="CodeMirror-line-like"
                                >
                                  <span>
                                    xxxxxxxxxx
                                  </span>
                                </pre>
                              </div>
                              <div
                                class="CodeMirror-measure"
                              />
                              <div
                                style="position: relative; z-index: 1;"
                              />
                              <div
                                class="CodeMirror-cursors"
                              />
                              <div
                                class="CodeMirror-code"
                                role="presentation"
                              />
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        style="position: absolute; height: 50px; width: 1px;"
                      />
                      <div
                        class="CodeMirror-gutters"
                        style="left: 0px;"
                      >
                        <div
                          class="CodeMirror-gutter CodeMirrorReact-linenumbers"
                        />
                        <div
                          class="CodeMirror-gutter CodeMirror-foldgutter"
                        />
                        <div
                          class="CodeMirror-gutter CodeMirror-linenumbers"
                          style="width: 1px;"
                        />
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </section>
          </div>
        </div>
      </section>
    </form>
  </div>
  <div
    class="bx--bmrg-modal-portal"
  />
</body>
`;
